apps/web/src/pages/workspace/[workspaceId]/setting.tsx

```diff
@@ -43,56 +43,60 @@ function useTabRouterSync(
     settingPanelValues.indexOf(queryCurrentTab as SettingPanel) === -1
   ) {
     setCurrentTab(settingPanel.General);
-    void router.replace({
+    return router.replace({
       pathname: router.pathname,
       query: {
         ...router.query,
         currentTab: settingPanel.General,
       },
     });
-    return;
   } else if (settingPanelValues.indexOf(currentTab as SettingPanel) === -1) {
     setCurrentTab(settingPanel.General);
-    void router.replace({
+    return router.replace({
       pathname: router.pathname,
       query: {
         ...router.query,
         currentTab: settingPanel.General,
       },
     });
-    return;
   } else if (queryCurrentTab !== currentTab) {
-    void router.replace({
+    return router.replace({
       pathname: router.pathname,
       query: {
         ...router.query,
         currentTab: currentTab,
       },
     });
-    return;
   }
+  return null;
 }
 
 const SettingPage: NextPageWithLayout = () => {
   const router = useRouter();
   const [currentWorkspace] = useCurrentWorkspace();
   const t = useAFFiNEI18N();
   const [currentTab, setCurrentTab] = useAtom(settingPanelAtom);
-  useEffect(() => {});
   const onChangeTab = useCallback(
     (tab: SettingPanel) => {
       setCurrentTab(tab as SettingPanel);
-      void router.push({
-        pathname: router.pathname,
-        query: {
-          ...router.query,
-          currentTab: tab,
-        },
-      });
+      router
+        .push({
+          pathname: router.pathname,
+          query: {
+            ...router.query,
+            currentTab: tab,
+          },
+        })
+        .catch(err => {
+          console.error(err);
+        });
     },
     [router, setCurrentTab]
   );
-  useTabRouterSync(router, currentTab, setCurrentTab);
+
+  useTabRouterSync(router, currentTab, setCurrentTab).catch(err => {
```

Comment:
@Himself65 this async function is called in render function, which is really weird

Reply:
I would like to wrap it into swr I think